src/extract-kindle-book.ts

```diff
@@ -259,7 +273,8 @@ async function main() {
     if (pageNav?.page === undefined) {
       break
     }
-    if (pageNav.page > totalContentPages) {
+    if (pageNav.page >= totalContentPages) {
```

Comment:
I noticed this in one of the books I extracted, but I had to do `(totalContentPages - 1)` otherwise it would keep trying to go to a last page that couldn't be navigated to. 

It's strange because the kindle app calculates the total content pages the same, but you can't navigate to that last page. It then gets stuck in a loop of screenshotting the same page and trying to navigate to the next.

With more time, a smarter solution would probably be checking if the current page has already been navigated to, then `break`

Reply:
yeah, in my testing it seemed to be slightly different for different books. i never had the chance to improve the robustness, but I like the idea of trying to get the current page ID and breaking if it's been visited